devito/ops/node_factory.py

```diff
@@ -0,0 +1,214 @@
+from sympy import Eq, Add, Mul, Pow, Integer, Float, Rational
+from devito import Dimension
+from devito.symbolics import Macro
+from devito.types import Symbol, Indexed, Array
+
+from devito.ops.utils import namespace
+
+
+class OpsNodeFactory():
+    """
+    Generates ops nodes for building an OPS expression.
+
+    Examples
+    --------
+    >>> a = OpsNodeFactory()
+    >>> b = a.new_symbol('symbol_name')
+    """
+
+    def __init__(self):
+        self.ops_grids = {}
+
+    def new_symbol(self, name):
+        """
+        Creates a new sympy :class:`Symbol` object with the given name.
+        Its used to define/use variables.
```

Comment:
It is(?) [define/use --> Is used necessary?],

Reply:
You are right. As the method will always create new object there is no point in the "use".